hpx-sys/src/lib.rs

```diff
@@ -39,9 +40,45 @@ pub mod ffi {
 // Wrapper for the above Bindings.
 // reffer to tests to understand how to use them. [NOTE: Not all bindings have wrapper.]
 // ================================================================================================
-use std::ffi::CString;
```

Comment:
@Connor-GH can you explain why have you removed `std::ffi::String`?
> I would really prefer not to pass this function pointer through by another static variable, but it was the only way I could find to keep the API of fn(Vec) to the user. It feels really bad, please work with me on fixing this.

- `CString` in Rust is a better approach, and it eliminate the need for the c_init function and the static variable.
- or ig instead of `static variable` you can use `closure` like the following
```
pub fn init<F>(func: F, func_args: Vec<String>) -> i32
where
    F: Fn(Vec<String>) -> i32 + 'static,
{
    let c_init = ......
    ....
    ...
    unsafe { self::ffi::init(c_init, argc, argv) }
}
```

Reply:
> can you explain why have you removed `std::ffi::String`?
It is right here: https://github.com/STEllAR-GROUP/hpx-rs/pull/15/files#diff-7ab105a9531556e61127ea7119050c01f680c4d93e2fbaa66cb1a4c19bbe66f7R44

Reply:
Please explain what you mean by "you can use ``closure``". A closure cannot be cast into a function pointer if it takes any arguments.